test/e2e/webhook_test.go

```diff
@@ -1291,3 +1293,80 @@ var _ = Describe("webhook tests for ResourceOverride UPDATE operations", Ordered
 		}, testutils.PollTimeout, testutils.PollInterval).Should(Succeed())
 	})
 })
+
+var _ = Describe("webhook tests for operations on ARM managed resources", Ordered, func() {
+	BeforeAll(func() {
+		By("creating a managed namespace")
+		createManagedNamespace()
+	})
+
+	AfterAll(func() {
+		By("deleting the managed namespace")
+		ns := managedNamespace()
+		Expect(hubClient.Delete(ctx, &ns)).Should(SatisfyAny(Succeed(), utils.NotFoundMatcher{}), "Failed to delete the managed namespace")
+		Eventually(
+			func() error {
+				if err := hubClient.Get(ctx, types.NamespacedName{Name: ns.Name}, &corev1.Namespace{}); !k8sErrors.IsNotFound(err) {
+					return fmt.Errorf("The managed namespace still exists or an unexpected error occurred: %w", err)
+				}
+				return nil
+			}, workloadEventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to remove the managed namespace %s", ns.Name)
+	})
+
+	It("should deny create a managed resource namespace", func() {
+		Eventually(func(g Gomega) error {
+			createNs := managedNamespace()
+			createNs.Name = "this-will-be-denied"
+			err := impersonateHubClient.Create(ctx, &createNs)
+			if k8sErrors.IsConflict(err) {
+				return err
+			}
+			var statusErr *k8sErrors.StatusError
+			g.Expect(errors.As(err, &statusErr)).To(BeTrue(), fmt.Sprintf("Create managed namespace call produced error %s. Error type wanted is %s.", reflect.TypeOf(err), reflect.TypeOf(&k8sErrors.StatusError{})))
+			Expect(statusErr.ErrStatus.Message).Should(MatchRegexp(".*the operation on the managed resource type .* is not allowed"))
+			return nil
+		}, testutils.PollTimeout, testutils.PollInterval).Should(Succeed())
+	})
+
+	It("should deny update a managed resource namespace", func() {
+		Eventually(func(g Gomega) error {
+			updateNamespace := managedNamespace()
+			updateNamespace.Labels["foo"] = "NotManaged"
+			err := impersonateHubClient.Update(ctx, &updateNamespace)
+			if k8sErrors.IsConflict(err) {
+				return err
+			}
+			var statusErr *k8sErrors.StatusError
+			g.Expect(errors.As(err, &statusErr)).To(BeTrue(), fmt.Sprintf("Update managed namespace call produced error %s. Error type wanted is %s.", reflect.TypeOf(err), reflect.TypeOf(&k8sErrors.StatusError{})))
+			Expect(statusErr.ErrStatus.Message).Should(MatchRegexp(".*the operation on the managed resource type .* is not allowed"))
+			return nil
+		}, testutils.PollTimeout, testutils.PollInterval).Should(Succeed())
+	})
+
+	It("should deny delete a managed resource namespace", func() {
+		Eventually(func(g Gomega) error {
+			created := managedNamespace()
+			err := impersonateHubClient.Delete(ctx, &created)
```

Comment:
Same for delete calls they don't need an Eventually block, and we can't conflict on a delete call